File: food-swipe-admin-backend-go/ingredient/service/create-ingredient.go
package service

import "food-swipe.app/ingredient/dto"

func (s *Service) CreateIngredient(payload *dto.CreateIngredientDto) (*Ingredient, error) {
	ingredient, err := s.storage.CreateIngredient(payload.Name)
	if err != nil {
		return nil, err
	}

	return storageIngredientToIngredient(ingredient), nil
}

File: packages/client/api/src/ingredient/api/create-ingredient.ts
import type { Ingredient } from "@food-swipe/types";
import type { HttpClient } from "../../client";

export type CreateIngredientInput = {
  data: {
    name: string;
  };
};

export const createIngredient = async (api: HttpClient, payload: CreateIngredientInput) => {
  const response = await api.fetch("/v1/ingredients", {
    body: JSON.stringify(payload.data),
    method: "POST",
  });
  return response.json() as Promise<Ingredient>;
};
